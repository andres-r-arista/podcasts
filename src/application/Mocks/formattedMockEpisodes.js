export const formattedMockEpisodes = [
  {
    trackId: 1000610181965,
    trackName: 'Episode 620 | "Who Said?"',
    description:
      "The gang records their first episode without Joe as they begin with the latest news on Jonathan Majors as additional abuse allegations come to light (12:47) and Tiffany Haddish attempts to defend him (25:57). Desiigner seeks mental health treatment after exposing himself on an airplane (36:55), Frank Ocean pulls out of weekend 2 of Coachella due to an injury (1:00:33), and *SPOILER ALERT* Snowfall series finale recap (1:05:00). Also, the JBP gives their thoughts on the Chance the Rapper dancing video (1:29:01), Eagles QB Jalen Hurts pens a love letter to black women (1:40:06), + MORE!\n  \n Become a Patron of The Joe Budden Podcast for additional bonus episodes and visual content for all things JBP.: Tap in here www.patreon.com/JoeBudden\n \n Sleeper Picks:\n Ice | Lloyd Banks - “Cliffhanger”\n Melyssa | SamTRax (feat. Amber Oliver) - “You Are”\n Parks | Your Old Droog & Madlib - “Pronouns”\n Ish | Yung Bleu & Chris Brown - “Distant Lover”",
    episodeUrl:
      "https://traffic.libsyn.com/secure/jbpod/Joe_Budden_Podcast_620.mp3?dest-id=2422538",
    trackTimeMillis: 8665000,
    releaseDate: "2023-04-22T07:00:00Z",
  },
  {
    trackId: 1000609597323,
    trackName: 'Episode 619 | "Hate to be Frank"',
    description:
      "The JBP starts this episode by recapping Coachella (26:45) and giving their thoughts on Frank Ocean’s performance and what went wrong with the set (30:26). Ice recaps the Lil Wayne concert he attended (55:55), Jonathan Majors PR & management teams have severed ties (1:11:25), and the gang has some updates on the YSL RICO case (1:29:54). Also, an A.I. version of a Drake & Weeknd song was leaked (1:42:46), Part of the Show (1:57:05), the NBA playoffs (2:16:22), + MORE! \n Become a Patron of The Joe Budden Podcast for additional bonus episodes and visual content for all things JBP.: Tap in here www.patreon.com/JoeBudden\n  \n Sleeper Picks:\n Joe | Dende - “Block Me”\n Ice | Mozzy - “FREE ALL THE LIFERS”\n Parks | Agony Aka Southside Tone - “Ricks Story”\n Ish | Maeta - “S(EX) [Sped Up]”\n Melyssa | Satin Jackets & Panama - “Back to Me”\n QueenzFlip | Stack Bundles - “Drunk or High” \n  ",
    episodeUrl:
      "https://traffic.libsyn.com/secure/jbpod/Joe_Budden_Podcast_619.mp3?dest-id=2422538",
    trackTimeMillis: 10192000,
    releaseDate: "2023-04-19T07:00:00Z",
  },
];
