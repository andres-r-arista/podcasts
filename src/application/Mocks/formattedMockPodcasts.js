export const formattedMockPodcasts = [
  {
    id: "1535809341",
    title: "The Joe Budden Podcast",
    author: "The Joe Budden Network",
    imgUrl:
      "https://is1-ssl.mzstatic.com/image/thumb/Podcasts113/v4/f2/21/fa/f221fabd-017f-5125-633b-f1fe4f39802a/mza_182995249085044287.jpg/170x170bb.png",
    description:
      "Tune into Joe Budden and his friends. Follow along the crazy adventures of these very random friends.",
  },
  {
    id: "1592180670",
    title: "The Prince Mixtape",
    author: "CNN",
    imgUrl:
      "https://is2-ssl.mzstatic.com/image/thumb/Podcasts116/v4/49/c2/11/49c211d2-629a-dab1-48e4-a522521f3ea1/mza_14590872972922522730.jpg/170x170bb.png",
    description:
      "On each episode of The Prince Mixtape, weâ€™ll meet people who knew Prince, worked with him, or simply loved him from afar. Host Nichole Perkins will dig deep into key moments of his life, like when his sexually liberated lyrics gave birth to the Parental Advisory Sticker, when he fought to own his masters, and when he rocked that infamous, delicious booty cut-out pantsuit. The first season of this series explored the unforgettable life of Princess Diana. On season two, just like a treasured mixtape from your favorite crush, the series is a reflection of all the care and passion, the big moments and small, that made up a life. Prince's life.",
  },
];
