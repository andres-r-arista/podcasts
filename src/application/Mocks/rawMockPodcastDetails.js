export const rawMockPodcastDetails = {
  resultCount: 3,
  results: [
    {
      wrapperType: "track",
      kind: "podcast",
      artistId: 1536636674,
      collectionId: 934552872,
      trackId: 934552872,
      artistName: "Vulture",
      collectionName: "Switched on Pop",
      trackName: "Switched on Pop",
      collectionCensoredName: "Switched on Pop",
      trackCensoredName: "Switched on Pop",
      artistViewUrl:
        "https://podcasts.apple.com/us/artist/vulture/1536636674?uo=4",
      collectionViewUrl:
        "https://podcasts.apple.com/us/podcast/switched-on-pop/id934552872?uo=4",
      feedUrl: "https://feeds.megaphone.fm/switchedonpop",
      trackViewUrl:
        "https://podcasts.apple.com/us/podcast/switched-on-pop/id934552872?uo=4",
      artworkUrl30:
        "https://is2-ssl.mzstatic.com/image/thumb/Podcasts116/v4/6e/7c/17/6e7c170e-ecd6-a572-a851-6b0a02a168c5/mza_7324103647670443393.jpeg/30x30bb.jpg",
      artworkUrl60:
        "https://is2-ssl.mzstatic.com/image/thumb/Podcasts116/v4/6e/7c/17/6e7c170e-ecd6-a572-a851-6b0a02a168c5/mza_7324103647670443393.jpeg/60x60bb.jpg",
      artworkUrl100:
        "https://is2-ssl.mzstatic.com/image/thumb/Podcasts116/v4/6e/7c/17/6e7c170e-ecd6-a572-a851-6b0a02a168c5/mza_7324103647670443393.jpeg/100x100bb.jpg",
      collectionPrice: 0.0,
      trackPrice: 0.0,
      collectionHdPrice: 0,
      releaseDate: "2023-04-18T09:00:00Z",
      collectionExplicitness: "notExplicit",
      trackExplicitness: "notExplicit",
      trackCount: 340,
      trackTimeMillis: 2940,
      country: "USA",
      currency: "USD",
      primaryGenreName: "Music Commentary",
      artworkUrl600:
        "https://is2-ssl.mzstatic.com/image/thumb/Podcasts116/v4/6e/7c/17/6e7c170e-ecd6-a572-a851-6b0a02a168c5/mza_7324103647670443393.jpeg/600x600bb.jpg",
      genreIds: ["1523", "26", "1310", "1525"],
      genres: ["Music Commentary", "Podcasts", "Music", "Music Interviews"],
    },
    {
      country: "USA",
      artworkUrl600:
        "https://is2-ssl.mzstatic.com/image/thumb/Podcasts116/v4/6e/7c/17/6e7c170e-ecd6-a572-a851-6b0a02a168c5/mza_7324103647670443393.jpeg/600x600bb.jpg",
      previewUrl:
        "https://www.podtrac.com/pts/redirect.mp3/pdst.fm/e/chtbl.com/track/524GE/traffic.megaphone.fm/VMP6807581673.mp3?updated=1682099892",
      episodeFileExtension: "mp3",
      artworkUrl160:
        "https://is2-ssl.mzstatic.com/image/thumb/Podcasts116/v4/6e/7c/17/6e7c170e-ecd6-a572-a851-6b0a02a168c5/mza_7324103647670443393.jpeg/160x160bb.jpg",
      episodeContentType: "audio",
      feedUrl: "https://feeds.megaphone.fm/switchedonpop",
      closedCaptioning: "none",
      collectionId: 934552872,
      collectionName: "Switched on Pop",
      artistIds: [1536636674],
      trackTimeMillis: 2234000,
      collectionViewUrl:
        "https://itunes.apple.com/us/podcast/switched-on-pop/id934552872?mt=2&uo=4",
      genres: [{ name: "Music Commentary", id: "1523" }],
      episodeGuid: "5ca3c8e2-e04d-11ed-b32b-ab0c3149d5a3",
      description:
        "We have an emergency podcast drop because the biggest and fastest moving story on the internet right now is about a song called “Heart On My Sleeve.” The track sounds like it was made by the producer Metro Boomin featuring Drake and The Weeknd. It might be one of the most consequential songs in music history because it was actually a fake, made with artificial intelligence. The blowback from this song has been enormous and a bit confusing. So host Charlie Harding went on The Vergecast podcast to break down how this song was likely made, and what it might mean for the music industry, the tech industry and all of our intellectual property. \n\n\nListen to the whole episode on The Vergecast\n\n\nRead Alex Cranz's story on Laser Bongs on The Verge\n\n\n\nLearn more about your ad choices. Visit podcastchoices.com/adchoices",
      releaseDate: "2023-04-21T14:16:00Z",
      trackId: 1000610099308,
      trackName: "A.I. Drake has put music in a tailspin",
      shortDescription: "Lawlessness and Laser Bongs with The Vergecast",
      artworkUrl60:
        "https://is2-ssl.mzstatic.com/image/thumb/Podcasts116/v4/6e/7c/17/6e7c170e-ecd6-a572-a851-6b0a02a168c5/mza_7324103647670443393.jpeg/60x60bb.jpg",
      artistViewUrl:
        "https://itunes.apple.com/us/artist/vulture/1536636674?mt=2&uo=4",
      episodeUrl:
        "https://www.podtrac.com/pts/redirect.mp3/pdst.fm/e/chtbl.com/track/524GE/traffic.megaphone.fm/VMP6807581673.mp3?updated=1682099892",
      trackViewUrl:
        "https://podcasts.apple.com/us/podcast/a-i-drake-has-put-music-in-a-tailspin/id934552872?i=1000610099308&uo=4",
      kind: "podcast-episode",
      wrapperType: "podcastEpisode",
    },
    {
      country: "USA",
      artworkUrl600:
        "https://is4-ssl.mzstatic.com/image/thumb/Podcasts126/v4/c6/f6/57/c6f657e3-6e84-6a9c-c1e4-f40d3e18c5e1/mza_868627494590202203.jpg/600x600bb.jpg",
      previewUrl:
        "https://www.podtrac.com/pts/redirect.mp3/pdst.fm/e/chtbl.com/track/524GE/traffic.megaphone.fm/VMP3036086010.mp3?updated=1681777869",
      episodeFileExtension: "mp3",
      artworkUrl160:
        "https://is4-ssl.mzstatic.com/image/thumb/Podcasts126/v4/c6/f6/57/c6f657e3-6e84-6a9c-c1e4-f40d3e18c5e1/mza_868627494590202203.jpg/160x160bb.jpg",
      episodeContentType: "audio",
      feedUrl: "https://feeds.megaphone.fm/switchedonpop",
      closedCaptioning: "none",
      collectionId: 934552872,
      collectionName: "Switched on Pop",
      artistIds: [1536636674],
      trackTimeMillis: 2940000,
      collectionViewUrl:
        "https://itunes.apple.com/us/podcast/switched-on-pop/id934552872?mt=2&uo=4",
      genres: [{ name: "Music Commentary", id: "1523" }],
      episodeGuid: "7b452ca0-3512-11ed-bde1-77534e550098",
      description:
        "The South Korean idol group BTS is one of the biggest musical sensations in history. They're constantly breaking records and they have one of the most dedicated fan bases in the world known as Army. Their hit singles like “Boy With Love,” “Dynamite,” and “Butter” have been discussed on the show in the past for breaking through the US charts. But back in 2022, they decided to take a break from group activities and start releasing solo material because of their obligations to each fulfill mandatory military service in South Korea, precluding them from working together at the same time. Solo projects aren’t new for the group – BTS members have released mixtapes as far back as 2015 and countless solo singles – but this was the first time that BTS had ever announced a prolonged break. This moment, originally presented as a hiatus, has evolved into a whole new musical chapter for the group, with a seemingly endless array of new solo releases. Switched On Pop listens back to one track from each solo effort so far to introduce the casual listener to what is happening in this new era of BTS. Joining the show is Lenika Cruz, senior editor at The Atlantic, who literally wrote the book on BTS, simply called On BTS out on the Atlantic Editions imprint.\n\nSongs Discussed:\n\nj-hope, J. Cole - on the street\n\nJimin - Like Crazy\n\nRM, Youjeen - Wild Flower\n\nAgust D - People Pt.2\n\nAgust D - People\n\nJung Kook - Dreamers\n\nJIN - The Astronaut\n\nBTS - Singularity, Inner Child\n\nColdplay, BTS - My Universe\n\n\nLearn more about your ad choices. Visit podcastchoices.com/adchoices",
      releaseDate: "2023-04-18T09:00:00Z",
      trackId: 1000609451169,
      trackName: "BTS goes solo together",
      shortDescription: "",
      artworkUrl60:
        "https://is4-ssl.mzstatic.com/image/thumb/Podcasts126/v4/c6/f6/57/c6f657e3-6e84-6a9c-c1e4-f40d3e18c5e1/mza_868627494590202203.jpg/60x60bb.jpg",
      artistViewUrl:
        "https://itunes.apple.com/us/artist/vulture/1536636674?mt=2&uo=4",
      episodeUrl:
        "https://www.podtrac.com/pts/redirect.mp3/pdst.fm/e/chtbl.com/track/524GE/traffic.megaphone.fm/VMP3036086010.mp3?updated=1681777869",
      trackViewUrl:
        "https://podcasts.apple.com/us/podcast/bts-goes-solo-together/id934552872?i=1000609451169&uo=4",
      kind: "podcast-episode",
      wrapperType: "podcastEpisode",
    },
  ],
};
