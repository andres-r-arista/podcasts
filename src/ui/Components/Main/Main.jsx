import React, { useState } from "react";

import PropTypes from "prop-types";

import { PodcastCard } from "../PodcastCard";
import { FilterBar } from "../FilterBar";

export const Main = ({ podcasts }) => {
  const [filteredPodcasts, setFilteredPodcasts] = useState(podcasts);
  const filterPodcasts = (event) => {
    setFilteredPodcasts(
      podcasts.filter(
        (podcast) =>
          podcast.title
            .toLowerCase()
            .includes(event.target.value.toLowerCase()) ||
          podcast.author
            .toLowerCase()
            .includes(event.target.value.toLowerCase())
      )
    );
  };
  return (
    <div data-testid="main">
      <FilterBar
          podcastsAmount={filteredPodcasts.length || 0}
          onChange={filterPodcasts}
        />
      <div className="flex flex-wrap">
        {filteredPodcasts && filteredPodcasts.map((podcast) => (
          <PodcastCard
            title={podcast.title}
            key={podcast.id}
            id={podcast.id}
            author={podcast.author}
            imgUrl={podcast.imgUrl}
          />
        ))}
      </div>
    </div>
  );
};

Main.propTypes = {
  podcasts: PropTypes.array.isRequired,
};