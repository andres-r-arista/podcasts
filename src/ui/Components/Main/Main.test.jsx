import { render, screen, fireEvent } from "@testing-library/react";

import { formattedMockPodcasts } from "../../../application/Mocks/formattedMockPodcasts";
import { Main } from "./Main";

const mockedUsedNavigate = jest.fn();
jest.mock("react-router-dom", () => ({
  ...jest.requireActual("react-router-dom"),
  useNavigate: () => mockedUsedNavigate,
}));

describe("Tests in <Main />", () => {
  test("must match the snapshot", () => {
    const { container } = render(<Main podcasts={formattedMockPodcasts} />);
    expect(container).toMatchSnapshot();
  });

  test("must have no podcast cards if receiving no podcasts", () => {
    render(<Main podcasts={[]} />);
    expect(screen.queryAllByTestId("podcast-card")).toHaveLength(0);
  });

  test("must have two podcast cards if receiving two podcasts", () => {
    render(<Main podcasts={formattedMockPodcasts} />);
    expect(screen.queryAllByTestId("podcast-card")).toHaveLength(2);
  });

  test("must have a filter bar", () => {
    render(<Main podcasts={formattedMockPodcasts} />);
    expect(screen.getByTestId("filter-bar"));
  });

  test("must filter the podcasts by title (case insensitive) when entering a string in the filter bar input", () => {
    render(<Main podcasts={formattedMockPodcasts} />);
    expect(screen.queryAllByTestId("podcast-card")).toHaveLength(2);
    const screenInput= screen.getByTestId("filter-bar-input");
    fireEvent.change(screenInput, { target: { value: 'the joe budden podcast' } });
    expect(screen.queryAllByTestId("podcast-card")).toHaveLength(1);
  });

  test("must filter the podcasts by author (case insensitive) when entering a string in the filter bar input", () => {
    render(<Main podcasts={formattedMockPodcasts} />);
    expect(screen.queryAllByTestId("podcast-card")).toHaveLength(2);
    const screenInput= screen.getByTestId("filter-bar-input");
    fireEvent.change(screenInput, { target: { value: 'the joe budden network' } });
    expect(screen.queryAllByTestId("podcast-card")).toHaveLength(1);
  });
});
