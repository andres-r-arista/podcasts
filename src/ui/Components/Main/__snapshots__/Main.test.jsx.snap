// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests in <Main /> must match the snapshot 1`] = `
<div>
  <div
    data-testid="main"
  >
    <div
      class="flex justify-end mr-4"
      data-testid="filter-bar"
    >
      <div
        class="px-1 rounded-lg self-center mr-2 bg-sky-800 text-white font-bold"
        data-testid="badge"
      >
        2
      </div>
      <input
        class="border text-sm px-2 py-1 w-64 rounded"
        data-testid="filter-bar-input"
        placeholder="Filter podcasts..."
        type="text"
      />
    </div>
    <div
      class="flex flex-wrap"
    >
      <div
        class="w-1/4"
        data-testid="podcast-card"
      >
        <div
          class="border text-center h-24 mx-4 my-16 shadow-md cursor-pointer"
          data-testid="podcast-card-box"
        >
          <div
            class="flex flex-col -translate-y-8 items-center"
          >
            <img
              alt="The Joe Budden Podcast"
              class="w-16 h-16 rounded-full"
              data-testid="podcast-card-image"
              src="https://is1-ssl.mzstatic.com/image/thumb/Podcasts113/v4/f2/21/fa/f221fabd-017f-5125-633b-f1fe4f39802a/mza_182995249085044287.jpg/170x170bb.png"
            />
            <h1
              class="uppercase font-semibold truncate max-w-full text-sm px-2"
              data-testid="podcast-card-title"
              title="The Joe Budden Podcast"
            >
              The Joe Budden Podcast
            </h1>
            <h2
              class="text-gray-400 font-semibold truncate max-w-full text-xs px-2"
              data-testid="podcast-card-author"
              title="The Joe Budden Network"
            >
              Author: The Joe Budden Network
            </h2>
          </div>
        </div>
      </div>
      <div
        class="w-1/4"
        data-testid="podcast-card"
      >
        <div
          class="border text-center h-24 mx-4 my-16 shadow-md cursor-pointer"
          data-testid="podcast-card-box"
        >
          <div
            class="flex flex-col -translate-y-8 items-center"
          >
            <img
              alt="The Prince Mixtape"
              class="w-16 h-16 rounded-full"
              data-testid="podcast-card-image"
              src="https://is2-ssl.mzstatic.com/image/thumb/Podcasts116/v4/49/c2/11/49c211d2-629a-dab1-48e4-a522521f3ea1/mza_14590872972922522730.jpg/170x170bb.png"
            />
            <h1
              class="uppercase font-semibold truncate max-w-full text-sm px-2"
              data-testid="podcast-card-title"
              title="The Prince Mixtape"
            >
              The Prince Mixtape
            </h1>
            <h2
              class="text-gray-400 font-semibold truncate max-w-full text-xs px-2"
              data-testid="podcast-card-author"
              title="CNN"
            >
              Author: CNN
            </h2>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
